view/src/bin/pviewd.rs

```diff
@@ -34,6 +36,49 @@ struct Opt {
     pd_port: u16,
 }
 
+impl Opt {
+    #[instrument(skip(self))]
+    pub async fn status(&self) -> Result<StatusResponse, anyhow::Error> {
+        let client = reqwest::Client::new();
+
+        let rsp: serde_json::Value = client
+            .get(format!(
+                r#"http://{}:{}/status"#,
+                self.node, self.tendermint_port
+            ))
+            .send()
+            .await?
+            .json()
+            .await?;
+
+        tracing::info!("{}", rsp);
+
+        let sync_info = rsp.get("sync_info").unwrap();
+
+        let sync_height = sync_info
+            .get("latest_block_height")
+            .and_then(|c| c.as_u64())
+            .ok_or_else(|| anyhow::anyhow!("could not parse JSON response"))?;
+
+        let synchronized = sync_info
+            .get("catching_up")
+            .and_then(|c| c.as_bool())
+            .ok_or_else(|| anyhow::anyhow!("could not parse JSON response"))?;
+
+        //TODO: how are sync_height & chain_height semantically different here? what does max_peer_block_height represent?
+        let chain_height = sync_info
+            .get("max_peer_block_height")
+            .and_then(|c| c.as_u64())
+            .ok_or_else(|| anyhow::anyhow!("could not parse JSON response"))?;
+
+        Ok(StatusResponse {
+            sync_height,
+            synchronized,
+            chain_height,
+        })
```

Comment:
We should change the `StatusResponse` to:
```proto
message StatusResponse {
    // The height the view service has synchronized to so far
    uint64 sync_height = 1;
    // Whether the view service is catching up with the chain state
    bool catching_up = 2;
}
```

Then `sync_height` is set to the sync height of the view service, and `catching_up` could be determined as follows:
```
let height_diff = latest_block_height.checked_sub(sync_height).ok_or_else(|| anyhow!("sync height ahead of node height"))?;
let catching_up = match (node_catching_up, height_diff) {
    // We're synced to the same height as the node
    (false, 0) => false,
    // We're one block behind, and will learn about it soon, close enough
    (false, 1) => false,
    // We're behind the node
    (false, _) => true,
    // The node is behind the network
    (true, _) => true,
};
```